skyportal/models/user_notification.py

```diff
@@ -291,6 +292,52 @@ def send_sms_notification(mapper, connection, target):
             log(f"Error sending sms notification: {e}")
 
 
+@event.listens_for(UserNotification, 'after_insert')
+def send_phone_notification(mapper, connection, target):
+    resource_type = notification_resource_type(target)
+    prefs = user_preferences(target, "phone", resource_type)
+    if not prefs:
+        return
+
+    sending = False
+    if prefs[resource_type]['phone'].get("on_shift", False):
+        current_shift = (
+            Shift.query.join(ShiftUser)
+            .filter(ShiftUser.user_id == target.user.id)
+            .filter(Shift.start_date <= arrow.utcnow().datetime)
+            .filter(Shift.end_date >= arrow.utcnow().datetime)
+            .first()
+        )
+        if current_shift is not None:
+            sending = True
+    else:
+        timeslot = prefs[resource_type]['phone'].get("time_slot", [])
+        if len(timeslot) > 0:
+            current_time = arrow.utcnow().datetime
+            if timeslot[0] < timeslot[1]:
+                if (
+                    current_time.hour >= timeslot[0]
+                    and current_time.hour <= timeslot[1]
+                ):
+                    sending = True
+            else:
+                if current_time.hour <= timeslot[0] or current_time.hour >= timeslot[1]:
```

Comment:
Isn't this supposed to be `and` as well?

Reply:
Actually no because this is the option where you are wanting calls outside of a specific time block.

Reply:
Then shouldn't the inequalities be flipped? 
In this case `timeslot[0]` is bigger than `timeslot[1]`. 
Lets assume `timeslot[0]=15` and `timeslot[1]=10`. 
If current time is 9 or 13, it is `True` because `current_time.hour<=timeslot[0]`. 
If current time is 13 or 17, it is `True` because `current_time.hour>=timeslot[1]`. 
I don't think there are any values where it is `False`. 
Unless I missed something this should also be fixed in the SMS notification.

Reply:
I want @Theodlz to weigh in here as he thought about it most.

Reply:
Guy is right! The 0 and the 1 need to be flipped in that else statement, it's a mistake. Please apply the same changes to the sms method so both are fixed.